src/Todo.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import './Todo.css';
+
+interface TodoItem {
+  todo: string;
+  completed: boolean;
+}
+
+const Todo = () => {
+  const [todo, setTodo] = useState<string>('');
+  const [array, setArray] = useState<TodoItem[]>([]);
+
+  const addTodo = () => {
+    if (todo.trim() !== '') {
+      setArray([...array, { todo, completed: false }]);
+      setTodo('');
+    } else {
+      alert('Enter task');
+    }
+  };
+  const handleInput = (str: string) => {
+    setTodo(str);
+  };
+
+  const deleteTodo = (index: number) => {
+    const newArray = [...array];
+    newArray.splice(index, 1);
+    setArray(newArray);
+  };
+
+  const Completion = (index: number) => {
+    const newArray = [...array];
+    newArray[index].completed = !newArray[index].completed;
+    setArray(newArray);
+  };
+
+  return (
+    <div className="todo-container">
+      <h1>Todo app</h1>
+      <input type="text" value={todo} onChange={(e) => handleInput(e.target.value)} />
```

Comment:
Instead of `onChange={(e) => handleInput(e.target.value)}`
you can do `onChange={handleInput}`
& inside handleInput function you can do event.target.value

The reason for this is if you use first way i.e your current implementation...It creates a new arrow function reference whenever your component re-renders.
I know it might be confusing for you right now...you will understand it slowly. For now just remember to avoid using arrow functions in JSX